### PR TITLE
Add logFileVerbosity to MSBuild and VSBuild tasks

### LINKED ISSUE #7823: Please generate and upload diagnostic VSBuild logs when System.Debug is true
It would be nice if the VSBuild task uploaded a diagnostic log when `System.Debug` is 'true', much like the VSTest@2 task does (see its vstest.ts source). This can be done with an additional task currently, but then the log filename doesn't line up (i.e. doesn't have the same ordinal prefix) and is something build administrators / developers have to do before a problem arises, vs. just kicking a new build with `System.Debug` = `true`.

### PR DESCRIPTION
**Pull request by heaths**
Similar to VSTest task. Fixes #7823.

### CODE REVIEW

File: Tasks/VSBuildV1/VSBuild.ps1 [hunk 1]
```diff
@@ -16,7 +19,8 @@ try {
     [bool]$maximumCpuCount = Get-VstsInput -Name MaximumCpuCount -AsBool
     [bool]$restoreNugetPackages = Get-VstsInput -Name RestoreNugetPackages -AsBool
     [bool]$logProjectEvents = Get-VstsInput -Name LogProjectEvents -AsBool
-    [bool]$createLogFile = Get-VstsInput -Name CreateLogFile -AsBool
+    [bool]$createLogFile = (Get-VstsInput -Name CreateLogFile -AsBool) -or $debug
+    [string]$logFileVerbosity = if ($debug) { "diagnostic" } else { Get-VstsInput -Name logFileVerbosity }
```
[the remark this answers is not in the document]
Comment (heaths) [reply Tasks/VSBuildV1/VSBuild.ps1]: Given the verbosity of spew when `System.Debug` is `true` - something you enable when even existing logging isn't obvious - I feel like people are really asking for everything. More often than not, I need diagnostic logs to see what properties and items are set to. When a user sets `System.Debug`, may as well give them everything so they can avoid a 3rd attempt with no clear way of changing detailed to diagnostic (have to actually change the build definition).

[the remark this answers is not in the document]
Comment (moswald) [reply Tasks/VSBuildV1/VSBuild.ps1]: That's fair. I don't often have to debug MSBuild failures at that level, so I'm not the expert. I just know the amount of spew from diagnostic >>> detailed.

**Comment by heaths** [comment]
My branch was quite a bit behind so I rebased and bumped the versions during the merge.

**Comment by heaths** [comment]
@moswald I had to fix a couple of tests, but for the common tests (wasn't aware I had to run those explicitly), the change was to fix the splatting which didn't trigger rerunning of the test suite. I tried closing and reopening the PR (that's worked in the past) but the PR build still won't trigger. Can you at least kick off a manual PR build?

**Comment by moswald** [comment]
I can't do it manually; our Azure DevOps host is down at the moment, there was an issue during deployment and then also during the rollback. When it comes back online, it should pick up your change for a new build.

**Pull request merged**